server/src/main/java/org/opensearch/extensions/action/ExtensionActionResponse.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions.action;
+
+import org.opensearch.action.ActionResponse;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+/**
+ * This class encapsulates the transport response from extension
+ *
+ * @opensearch.internal
+ */
+public class ExtensionActionResponse extends ActionResponse {
+    private byte[] responseBytes;
+
+    ExtensionActionResponse(byte[] responseBytes) {
+        this.responseBytes = responseBytes;
+    }
+
+    ExtensionActionResponse(StreamInput in) throws IOException {
+        super(in);
+        responseBytes = in.readByteArray();
+    }
+
+    public byte[] getResponseBytes() {
+        return responseBytes;
+    }
+
+    public void setResponseBytes(byte[] responseBytes) {
+        this.responseBytes = responseBytes;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeByteArray(responseBytes);
```

Comment:
Nit: I know `super.writeTo(out)` is a no-op in responses (unlike requests) so it's technically not needed, but it's there precisely to pair it with `super(in)` and saves reviewer (me) time thinking about it and worrying about implementation details.

Reply:
There is no `super.writeTo` defined in `ActionResponse`, `TransportResponse`etc. Its abstract and is being implemented first here in the hierarchy. Am I missing something ?

Reply:
Well, I'm misremembering something apparently. I swear I read a comment somewhere about this.  But the gist of my comment is:
 - We are calling `super(in)` on the constructor.  This currently inherits a no-op.
 - We are not calling `super.writeTo()` when sending this.

The problem would/could occur if at some point someone changes the `TransportMessage` constructor from a no-op to actually do something, and it would break this asymmetric pair (and a lot of others).

My suggestion is to either do both, or do neither (i.e., don't do `super(in)`).

Reply:
Sure.